docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,90 @@
+Log Centralisation
+==================
+
+Context
+-------
+
+MetalK8s value is to provide, out of the box, some services in order to ease
+the monitoring and operation of the platform as well as workload running on top
+of it.
+
+MetalK8s provides monitoring service with Prometheus and AlertManager in
+order to expose system performance time series and alerts.
+
+The logs generated by the platform and the workloads constitute an essential
+piece of information that can be very helpful when it comes to understand the
+root cause of a failure or a performance degradation. Because of the
+distributed nature of MetalK8s and workloads running of top of it, it is
+important to have some tooling to collect and aggregate all logs in order to
+make them browsable from a central endpoint. It is also important to ease
+browsing and correlation with other type of indicators such as KPIs and alerts.
+
+In addition to the application logs, the platform and workload are also
+producing some audit logs. If the same system is used to manage application and
+audit logs, then it is important to provide the end users of the workloads
+some capabilities to secure and query the audit logs belonging to their usage.
```

Comment:
to query is enough. I wanted to say that a tenant doesn't want his audit logs to be consumable by another tenant. By the way, I have no idea if we want to use Loki for audit logs, maybe I should completely remove this part.

Reply:
Yes, the requirements for audit logs on various components would need another design, because they do not serve the same purpose (the ones we consider here are there for monitoring/troubleshooting, whereas audit can have many other uses).

Reply:
I think that audit-logs are important to be collected (it is a frequent customer request) and since we are providing a multi-tenancy solution, each tenant should be able to query the audit-logs related to his/her tenant, i.e. which user in my tenant did what, who/when were changes done to my tenant (e.g. changing quotas, resetting passwords etc.). I cannot comment on which system to use or if that is more application, i.e. Zenko related and hence should be handled somewhere else but I think in general audit-logs/trails is something we need to provide for our "overall" solution.

Reply:
This is why audit logs are not logs as (IMHO) we're discussing here.

'Log aggregation' is about logs an operator uses to debug/monitor/... some software.

Audit logs are consumed by a *user* of said software (or some third-party solution this *user* has in place to analyze said logs).

A *user* doesn't care about, say, `Unable to connect to Zookeeper, retrying in 5s`, this is for operators. Similarly, an *operator* shouldn't be inspecting things like `User XYZ has created bucket ABC`. Heck, `XYZ` and `ABC` may be PII this operator is not even supposed/allowed to see! However, the *user* (of this account, for example) should be able to inspect this.

Please don't merge the two concepts.